mslib/mscolab/server.py

```diff
@@ -844,13 +845,16 @@ def acs_post_handler(url):
                         email = attributes.get("email")
                         username = attributes.get("givenName")
 
-                        token = generate_confirmation_token(email)
+                        if email is not None and username is not None:
+                            token = generate_confirmation_token(email)
+                        else:
+                            render_template('errors/403.html'), 403
 
                     except (NameError, AttributeError, KeyError):
                         render_template('errors/403.html'), 403
```

Comment:
I think you could just use `attributes["email"]` and `attributes["givenName"]`, since you catch the KeyError exception already. That would shorten this a bit and not duplicate the `render_template` call.

Something like:
```python
                        email = attributes["email"]
                        username = attributes["givenName"]

                        token = generate_confirmation_token(email)
                    except (NameError, AttributeError, KeyError):
                        render_template('errors/403.html'), 403
```

Reply:
Thanks :)